openlibrary/plugins/upstream/borrow.py

```diff
@@ -566,26 +564,12 @@ def get_edition_loans(edition):
 
 def get_loan_link(edition, type):
     """Get the loan link, which may be an ACS4 link or BookReader link depending on the loan type"""
-    global content_server
-
     resource_id = edition.get_lending_resource_id(type)
 
     if type == 'bookreader':
         # link to bookreader
         return (resource_id, get_bookreader_stream_url(edition.ocaid))
 
-    if type in ['pdf','epub']:
-        # ACS4
-        if not content_server:
-            if not config.content_server:
-                # $$$ log
-                return None
-            content_server = lending.ContentServer(config.content_server)
-
-        if not resource_id:
-            raise Exception('Could not find resource_id for %s - %s' % (edition.key, type))
-        return (resource_id, content_server.get_loan_link(resource_id))
-
```

Comment:
@mekarpeles: Is `type` guaranteed to be `'bookreader'` in all non-error cases?

Reply:
There could still be a case somehow that epub or pdf (acs) types are triggered by some logic, but if that's the case, we want it to fail (and the logs should surface any case we've possible missed). The alternative is the service writ large hanging on acs dependencies (and given OL isn't capable of serving acs in its current state, this seems like the right approach). What we could be doing is returning sensible error messages in the event of a `type` other than `bookreader`, if we find any pathways still exist.